internal/format/base_visitor.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package format
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"text/scanner"
+
+	"github.com/emicklei/proto"
+	"github.com/uber/prototool/internal/text"
+)
+
+type baseVisitor struct {
+	*printer
+
+	Failures []*text.Failure
+}
+
+func newBaseVisitor(indent string) *baseVisitor {
+	return &baseVisitor{printer: newPrinter(indent)}
+}
+
+//func (v *baseVisitor) VisitMessage(element *proto.Message)         {}
+//func (v *baseVisitor) VisitService(element *proto.Service)         {}
+//func (v *baseVisitor) VisitSyntax(element *proto.Syntax)           {}
+//func (v *baseVisitor) VisitPackage(element *proto.Package)         {}
+//func (v *baseVisitor) VisitOption(element *proto.Option)           {}
+//func (v *baseVisitor) VisitImport(element *proto.Import)           {}
+//func (v *baseVisitor) VisitNormalField(element *proto.NormalField) {}
+//func (v *baseVisitor) VisitEnumField(element *proto.EnumField)     {}
+//func (v *baseVisitor) VisitEnum(element *proto.Enum)               {}
+//func (v *baseVisitor) VisitComment(element *proto.Comment)         {}
+//func (v *baseVisitor) VisitOneof(element *proto.Oneof)             {}
+//func (v *baseVisitor) VisitOneofField(element *proto.OneOfField)   {}
+//func (v *baseVisitor) VisitReserved(element *proto.Reserved)       {}
+//func (v *baseVisitor) VisitRPC(element *proto.RPC)                 {}
+//func (v *baseVisitor) VisitMapField(element *proto.MapField)       {}
+//func (v *baseVisitor) VisitGroup(element *proto.Group)             {}
+//func (v *baseVisitor) VisitExtensions(element *proto.Extensions)   {}
```

Comment:
Do we need to implement the `emicklei/proto.Visitor` interface here, or can we remove these?

Reply:
Done